src/pages/api/shorten.ts

```diff
@@ -4,9 +4,12 @@ import '@/types/cloudflare';
 
 export const runtime = 'edge';
 
-function isValidUrl(url: string): boolean {
-  const urlPattern = /^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/;
-  return urlPattern.test(url);
+function encodeUrl(url: string): string {
```

Comment:
should we add unit tests for this function?

Reply:
Do we need unit tests to test built in features 🤣

Reply:
I guess not, we can cross that bridge later if we need to

Reply:
```js
const message = 'http://1.2.3.4:8080' // Try edit me

console.log(new URL(message).toString())
console.log(new URL('Invalid').toString())
```

```sh
http://1.2.3.4:8080/

TypeError: URL constructor: Invalid is not a valid URL.
```